portals/publisher/src/main/webapp/source/src/app/components/Apis/Details/Endpoints/Prototype/MockImplEndpoints.jsx

```diff
@@ -9,115 +9,796 @@
+                    setMockConfig((prev) => ({
+                        ...prev, config: {
+                            ...prev.config,
+                            simulationDetails: {
+                                ...prev.config.simulationDetails,
+                                api: {
+                                    latency: latencySim,
+                                    error: errorSim,
+                                },
+                            },
+                        },
+                    }));
+                } else { // Method-level config
+                    setMockConfig((prev) => ({
+                        ...prev,
+                        config: {
+                            ...prev.config,
+                            simulationDetails: {
+                                ...prev.config.simulationDetails,
+                                [path]: {
+                                    ...(prev.config.simulationDetails?.[path] || {}),
+                                    [method]: {
+                                        latency: latencySim,
+                                        error: errorSim,
+                                    },
+                                },
+                            },
+                        },
+                    }));
+                }
+            }
+        }
+    };
 
     useEffect(() => {
-        api.getGeneratedMockScriptsOfAPI(api.id)
-            .then((response) => {
+        const fetchMockScripts = async () => {
+            try {
+                setSaveDisable(true)
+                const response = await api.getGeneratedMockScriptsOfAPI(api.id);
+                const tempNullScripts = []
+                response.obj.list = response.obj.list.map((methodObj) => {
+                    const { content, simulationScript } =
+                        splitContentAndSimulation(methodObj.content);
+                    setSimulationConfig(simulationScript, methodObj.path, methodObj.verb);
+                    if (content.trim().length === 0) {
+                        const newEntry = `${methodObj.verb.toLowerCase()} - ${methodObj.path}`;
+                        tempNullScripts.push(newEntry);
+                    }
+                    return {
+                        ...methodObj,
+                        content,
+                        simulationScript
+                    };
+                });
+                setNullScripts(tempNullScripts);
                 setMockScripts(response.obj.list);
-            })
-            .catch((e) => {
-                console.error(error);
-                Alert.error(`Something went wrong while fetching example mock scripts!!`);
+                if (tempNullScripts.length !== response.obj.list.length) {
+                    setIsFirstTimeSelection(false);
+                    setSaveDisable(false);
+                }
+            } catch (e) {
+                console.error(e);
```

Comment:
Avoid printing raw errors or error messages in production unless you're certain they do not contain sensitive information. If logging is necessary, prefer custom, sanitized messages that you've explicitly defined. Also, note that using console.log and console.error in production is generally discouraged. Please review other parts of the PR for similar patterns.